hack/release.sh

```diff
@@ -14,6 +14,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Documentation about this script and how to use it can be found
+# at https://github.com/knative/test-infra/tree/master/ci
```

Comment:
@adrcunha I would prefer to change the link into https://github.com/knative/test-infra/blob/master/ci/manual_release.md for release instruction especially.

Reply:
I prefer pointing to a more generic location than getting this outdated when things change in test-infra. Furthermore, manual releases are exceptions and normally should not be used.

Reply:
@adrcunha  Could you do a rebase removing the conflict?

Reply:
Done, thanks for the heads up.